frontend/app/(defaultLayout)/Troubleshooting/[device]/hooks/useTroubleshootingProblemsProps.ts

```diff
@@ -0,0 +1,60 @@
+import { WithProvidersProps } from '@components/common';
+import type { WithLayoutProps } from '@layouts/default/server';
+import { FaIconProps } from '@ifixit/icons';
+
+export type TroubleshootingProblemsApiData = {
+   editUrl: string;
+   historyUrl: string;
+   answersUrl: string;
+   deviceGuideUrl?: string;
+   devicePartsUrl?: string;
+   breadcrumbs: BreadcrumbEntry[];
+   id: number;
+   problems: Problems[];
+   answers: Answers[];
+   allAnswersUrl: string;
+   allAnswersCount: number;
```

Comment:
Can we replace these with `TroubleshootingAnswersData`? Are they meant to stay in sync?

Reply:
What's the syntax for that? Something like:

```jsx
export type TroubleshootingProblemsApiData = {
   editUrl: string;
   historyUrl: string;
   answersUrl: string;
   deviceGuideUrl?: string;
   devicePartsUrl?: string;
   breadcrumbs: BreadcrumbEntry[];
   id: number;
   problems: Problems[];
   answers: TroubleshootingAnswersData['answers'];
   allAnswersUrl: TroubleshootingAnswersData['allAnswersUrl'];
   allAnswersCount: TroubleshootingAnswersData['allAnswersUrl']
   title: string;
};
```

Reply:
I was thinking more like this:

```ts
export type TroubleshootingProblemsApiData = {
   editUrl: string;
   historyUrl: string;
   answersUrl: string;
   deviceGuideUrl?: string;
   devicePartsUrl?: string;
   breadcrumbs: BreadcrumbEntry[];
   id: number;
   problems: Problems[];
   title: string;
} & TroubleshootingAnswersData;
```

Reply:
Much cleaner, thanks! https://github.com/iFixit/react-commerce/pull/2205/commits/8e2e176233f850eb403a78f73b748d4c7dfb5f47